### ISSUE TITLE
How to use proxies?

### ISSUE THREAD

**Issue by yovanoc** [axios/axios#1680]
Hello I tried to use proxies but all i tried was unsuccessful, if someone can help me.

With the https module with node all works well but axios doesn't work (it returns my ip and not proxy one)

```js
import HttpsProxyAgent from "@/utils/proxy/https";
import axios, { AxiosRequestConfig } from "axios";
import * as https from "https";

const proxy = "http://186.193.186.3:20183";
const agent = new HttpsProxyAgent(proxy);
const opts: https.RequestOptions = {
  agent,
  host: "ip.jsontest.com",
  method: "POST",
  path: "/",
  port: 80,
  timeout: 10000
};
const post_req = https.request(opts, res => {
  res.setEncoding("utf8");
  res.on("data", chunk => {
    console.log("Response: " + chunk);
  });
});
// post_req.write("name=john");
post_req.end();

const config: AxiosRequestConfig = {
  baseURL: "http://ip.jsontest.com:80",
  httpsAgent: agent,
  proxy: false
};

const myaxios = axios.create(config);
myaxios.get("/").then(res => console.log("xxx res", res.data));
```
Response:

```
xxx res {ip: "MY IP HERE"}
Response: {"ip": "186.193.186.3"}
```

**Comment by cyberwombat** [comment]
I've had a similar issue and can confirm that this method https://github.com/axios/axios/issues/847#issue-222517050 works. I am not able to use an https proxy but have it working with http.  I am using hookbin.com for testing and it correctly shows my proxy URL.

**Comment by BooleanInTheKitchen** [comment]
proxies could be tricky with axios, currenlty only working ones are username/pass authenticated ones for me.

**Comment by chinesedfan** [comment]
You have set `proxy: false`.

```
const config: AxiosRequestConfig = {
  baseURL: "http://ip.jsontest.com:80",
  httpsAgent: agent,
  proxy: false
};
```

**Comment by cyberwombat** [comment]
@chinesedfan it's been a while but I seem to remember you need proxy:false in order to use HttpsProxyAgent - that's not the issue we are or were having.

**Comment by chinesedfan** [comment]
@cyberwombat Thanks. I think I need more learning about axios proxy.

**Comment by adam-s** [comment]
Using node-tunnel (https://github.com/koichik/node-tunnel) to create an agent also works. 

````
const tunnel = require('tunnel');
class ApiService {
  get proxyRequest() {
    const agent = tunnel.httpsOverHttp({
      proxy: {
        host: 'http://proxy.example.com',
        port: 22225,
        proxyAuth: `username:password`,
      },
    });
    return axios.create({
      agent,
    })
  }
}
````

**Comment by jasonsaayman** [comment]
Closed please see #2072
